aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -56,6 +57,9 @@ SparseTensor& zero_sparse_(SparseTensor& self) {
     _get_sparse_impl(self)->values().resize_({0});
   }
   _get_sparse_impl(self)->set_nnz(0);
+#else
+  at::empty_out(self, _get_sparse_impl(self)->sizes());
```

Comment:
Calling `zeros_out` actually makes it into an infinite recursion because that will in turn call into this function. I am actually a bit unsure what's the right function to call here (maybe `resize_and_clear_`?)